frontend/src/app/modules/home/projects/edit/project-alert/project-alert.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { BaseComponent } from '@core/components/base/base.component';
+import { AlertSettings } from '@shared/models/alert-settings/alert-settings';
+import { Project } from '@shared/models/projects/project';
+import { DialogService } from 'primeng/dynamicdialog';
+import { AlertCategory } from '@shared/models/alert-settings/alert-category';
+import { Data } from '../../data';
+
+@Component({
+    selector: 'app-project-alert',
+    templateUrl: './project-alert.component.html',
+    styleUrls: ['../edit.component.sass'],
+    providers: [Data, DialogService]
+})
+export class ProjectAlertComponent extends BaseComponent implements OnInit {
+    @Input() alertSetting = {} as AlertSettings;
+    @Input() editFormAlert: FormGroup = new FormGroup({});
+    @Input() project: Project;
+
+    alertCategoryValue = this.alertData.alertCategories;
+    specialAlertTypeValue = this.alertData.alertTypes;
+    alertTimeIntervalValue = this.alertData.alertTimeIntervals;
+    specialTypes = AlertCategory.Special;
```

Comment:
Why do you need all these fields? You can just use alertData

Reply:
Fix, use alertData